File: libro/121_factorial_recursivo.py
# Programa: FactorialRecursivo
# Descripción:
#   Este programa calcula el factorial de los primeros
#   números naturales, de forma recursiva

def factorial_recursivo(n):
    if (n <= 1):
        return 1
    else:
        return n * factorial_recursivo(n-1)

if __name__ == "__main__":

    for i in range(1,11):
        print("{:2d}! vale:{:10d}".format(i, factorial_recursivo(i)))

File: libro/121_factorial_recursivo.cpp
/* ********************************************************
* Programa: FactorialRecursivo
*
* Descripción:
*   Este programa calcula el factorial de los primeros
*   números naturales, de forma recursiva
******************************************************** */

#include <stdio.h>

int FactorialRecursivo( int n ) {
  if (n <= 1) {
    return 1;
  } else {
    return n * FactorialRecursivo( n-1 );
  }
}

int main() {

  for (int i =  0; i <= 10; i++) {
    printf( "%2d! vale:%10d\n", i, FactorialRecursivo( i ) );
  }
}
